config.wyam

```diff
@@ -845,6 +852,32 @@ public class FeedItemData
     }
 }
 
+public IEnumerable<IDocument> GetOpmlData(IReadOnlyList<IDocument> inputs, IExecutionContext context)
+{
+    return inputs.Select(
+        input => {
+            string feed = input.String("Feed");
+            string title = input.String("Title");
+
+            if (string.IsNullOrWhiteSpace(title)) {
+                // process the base Url as the title in the absence of one
+                var slashes = "//";
+                
+                int from = feed.IndexOf(slashes) + slashes.Length;
+                int to = feed.IndexOf("/", from);
+
+                title = feed.Substring(from, to - from);
+            }
+
+            MetadataItems metadata = new MetadataItems(){
+                { "Title", title },
+                { "Feed", feed }

```

Comment:
Since "Feed" isn't being given a new value you can omit it from the `MetadataItems` - the existing value from the `input` document will "fall through" and still be available in the new document.